### PR TITLE
Add Espanso 2.1.4

### PR DESCRIPTION
**Pull request by lkrms**
The developer has explained the beta tag in v2.1.4's release notes:

> With this release, the v2 will officially become the stable version,
> while version 0.7.3 (legacy version) will be deprecated. This
> decision comes after months of testing, and we're confident the v2 is
> ready for general adoption!

**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --new-cask <cask>` worked successfully.
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

Further to the above, it is my understanding that this release meets the "But There Is No Stable Version!" criteria, and I can personally attest that despite officially being a "beta" product since I started relying on it in late 2019, Espanso has been more stable than most of the "stable" software I use.

### CODE REVIEW

File: Casks/espanso.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  homepage "https://espanso.org/"
+
+  livecheck do
+    url "https://github.com/federico-terzi/espanso/releases/latest"
+    strategy :page_match
+    regex(%r{href=.*?/v(\d+(?:\.\d+)*)(?:-beta)?/Espanso-Mac-(?:Intel|M1)\.zip}i)
+  end
+
+  conflicts_with formula: "espanso"
+
```
Comment (bevanjkay) [inline Casks/espanso.rb:25]: @lkrms There is no formula with this token, hence why CI is currently failing.

**Review by bevanjkay — CHANGES_REQUESTED** [review]

File: Casks/espanso.rb (hunk 1 above)
Comment (lkrms) [reply Casks/espanso.rb:25]: Thanks for letting me know. This cask would deprecate a legacy formula provided by the developer in their own tap. Would including the tap prefix work here, or should I just remove the directive?

Comment (lkrms) [reply Casks/espanso.rb]: After finding existing `conflicts_with` declarations that used explicit tap references, I did the same, updated the cask to 2.1.5 and rebased. Hopefully CI will pass now. Thanks again.

Comment (lkrms) [reply Casks/espanso.rb]: Third-party taps don't seem to be supported by CI, so I've removed `conflicts_with` entirely. The cookbook indicates it hasn't been implemented for casks yet anyway, so this isn't much of a loss.

Comment (lkrms) [reply Casks/espanso.rb]: Hi @bevanjkay, just checking in to see if there's something more I should be doing here? Thank you.

**Review by bevanjkay — APPROVED** [review]

**Pull request merged**